streamparse/cli/remove_logs.py

```diff
@@ -36,19 +53,31 @@ def subparser_hook(subparsers):
                                       description=__doc__,
                                       help=main.__doc__)
     subparser.set_defaults(func=main)
+    subparser.add_argument('-A', '--remove_all_artifacts',
+                           help='Remove not only topology-specific logs, but '
+                                'also any other files for the topology in its '
+                                'workers-artifacts subdirectories.',
+                           action='store_true')
     add_environment(subparser)
     add_name(subparser)
+    add_override_name(subparser)
     add_pattern(subparser)
+    add_pool_size(subparser)
     subparser.add_argument('-u', '--user',
-                           help="User argument to sudo when deleting logs.")
+                           help="User argument to sudo when deleting logs.",
+                           default='root')
```

Comment:
I could see this being `env.user` instead of `root`. While `root` would be available to the end-user in most small installations and individual users, it's requesting significantly more privilege than may be required. In an environment where end-users do not have root access, this default would fail saying you don't have permissions to sudo rather than do not have permissions to delete file. So to be usable and not require more privileges by default, we may want to default to `env.user` and have the end-user specify e.g. `storm` or `root` if it didn't match their `env.user` for SSHing.

On the flip side, if we set `env.user` as the default, the first time someone in a small environment with full privileges gets an error they could change `env.user` to be `root` instead of passing `-u` all of the time, which would completely defeat any security.

Reply:
So there are actually two problems with that suggestion:

1.  `env` isn't activated when the argument gets added to the subparser, so we wouldn't be able to specify `env.user` as the default.
2.  Like you said, if we make it use `env.user`, you're going to have someone set their SSH user to `root` or something equally terrible. We currently use `sudo` without specifying a user, which will use `root` by default, so I was just making that explicit in case someone wanted to use a different user if they didn't have permissions for sudoing as root.